### PR TITLE
feat: add a hook to avoid we miss make document

### PR DESCRIPTION
**Pull request by chuang8511**
Because

- we sometimes forgot to make gen-doc to update document

This commit

- add pre-commit hook to avoid missing updating document

### CODE REVIEW

File: .hooks/chech_mdx_changes.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
```
Comment (donch1989) [inline .hooks/chech_mdx_changes.sh:1]: File name is wrong.

Comment (donch1989) [reply .hooks/chech_mdx_changes.sh:1]: And do we need this script?
I think we can just always run `make build-doc` and `make gen-doc` directly in the hooks.

Comment (chuang8511) [reply .hooks/chech_mdx_changes.sh]: [tested](https://github.com/instill-ai/component/pull/251/files)

Comment (chuang8511) [reply .hooks/chech_mdx_changes.sh]: can we merge this because it is not user-face PR.

**Review by jvallesm — APPROVED** [review]

**Pull request merged**